src/IB/IBFEMethod.cpp

```diff
@@ -1428,14 +1424,60 @@ IBFEMethod::initializeFEEquationSystems()
             }
 
             auto& F_system = equation_systems.add_system<ExplicitSystem>(FORCE_SYSTEM_NAME);
-            F_system.add_vector("tmp", /*projections*/ false, /*type*/ PARALLEL);
             for (unsigned int d = 0; d < NDIM; ++d)
             {
                 std::ostringstream os;
                 os << "F_" << d;
                 F_system.add_variable(os.str(), d_fe_order[part], d_fe_family[part]);
             }
         }
+
+        // Setup cached system vectors.
+        //
+        // NOTE: libMesh does not appear to preserve the type of the vector
+        // after restart, and so we manually reset these vectors here.
+        auto& X_system = equation_systems.get_system<ExplicitSystem>(COORDS_SYSTEM_NAME);
+        std::unique_ptr<NumericVector<double> > X_new_clone, X_half_clone;
+        if (from_restart)
+        {
+            NumericVector<double>* X_new = X_system.request_vector("new");
+            if (X_new) X_new_clone = X_new->clone();
+            NumericVector<double>* X_half = X_system.request_vector("half");
+            if (X_half) X_half_clone = X_half->clone();
+        }
+        X_system.remove_vector("new");
+        X_system.remove_vector("half");
+        X_system.add_vector("new", /*projections*/ true, /*type*/ GHOSTED);
+        X_system.add_vector("half", /*projections*/ true, /*type*/ GHOSTED);
+        if (X_new_clone) X_new_clone->localize(*X_system.request_vector("new"));
+        if (X_half_clone) X_half_clone->localize(*X_system.request_vector("half"));
```

Comment:
I may have internalized the wrong definition of `localize()`.

I thought that this basically copied from the source vector to the destination vector in a way that also handled changes in data layout (e.g. `PARALLEL` to `GHOSTED`).

Reply:
To the best of my knowledge localize copies everything to the current processor. I'll check.

Reply:
(I think that is what [this code](https://libmesh.github.io/doxygen/classlibMesh_1_1PetscVector.html#aa4890584d57c1e2f9b1fd09051b1b221) does? Although, as we have discussed, it looks like every MPI process will have a full copy of the `IS` indices, which is not great.)

Reply:
There are different definitions of `localize()`. Some do copy all data to the processor, but there is a specialized version when you are localizing to a `PetscVector` that uses `VecScatter` --- see above.

Reply:
Of course, I am OK using something other than `localize()`. We could use the new batch copy and ghost fill routines if you prefer.

Reply:
The tricky part about that code is line 671: that sets up the index range to contain everything (since `PetscVector::size()` calls `VecGetSize`, which is the global size), so the `IS` object fed into `VecScatter` contains everything :(

Reply:
I think I know how to set this up without localization. I'll put up a patch when I am done.

Reply:
This is harder than I thought it would be :(